cli/fmt.rs

```diff
@@ -32,38 +34,47 @@ fn get_config() -> dprint::configuration::Configuration {
   ConfigurationBuilder::new().deno().build()
 }
 
-fn check_source_files(
+async fn check_source_files(
   config: dprint::configuration::Configuration,
   paths: Vec<PathBuf>,
 ) -> Result<(), ErrBox> {
-  let mut not_formatted_files = vec![];
-  let formatter = dprint::Formatter::new(config);
-
-  for file_path in paths {
-    let file_path_str = file_path.to_string_lossy();
-    let file_contents = fs::read_to_string(&file_path)?;
-    let r = formatter.format_text(&file_path_str, &file_contents);
-    match r {
-      Ok(formatted_text) => {
-        if formatted_text != file_contents {
-          not_formatted_files.push(file_path);
+  let not_formatted_files_count = Arc::new(AtomicUsize::new(0));
+  let formatter = Arc::new(dprint::Formatter::new(config));
+  let output_lock = Arc::new(Mutex::new(0)); // prevent threads outputting at the same time
```

Comment:
What is the proper/standard way to do this in Rust? Right now I just through a dummy value (0) into the mutex then do `let _ = output_lock.lock().unwrap();` below to prevent multiple threads from outputting to the console at the same time. I feel like there should be something better?

Reply:
Maybe [a mpsc channel](https://doc.rust-lang.org/std/sync/mpsc/fn.sync_channel.html)? Each formatting thread would get a sender clone and main thread would consume messages as they become available.

Reply:
Ooops, this is async, so probably mpsc channel from `futures` crate :)